### PR TITLE
Added scss to gulp file, converted some rules to mixins, and set up s…

### PR DESCRIPTION
**Pull request by dhagandev**
…css with inheritance hierarchy

### CODE REVIEW

File: src/css/styles.css [hunk 1]
```diff
@@ -0,0 +1,86 @@
+.button {
```
Comment (kev-chan) [inline src/css/styles.css]: add whitespace and put end parenthesis on new lines

File: gulpfile.js [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
 });
 
 gulp.task('dev', function() {
-	connect.server({
-		root: 'src',
-		port: 9000
-	});
-});
\ No newline at end of file
+    exec('gulp sass');
```
Comment (kev-chan) [inline gulpfile.js]: there is a better way in gulp to have dependent processes run.  Use that instead of exec

File: src/scss/styles.scss [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    font-family: $family;
+    font-size: $size;
+    font-weight: $weight;
+    font-style: $style;
+    color: $color;
+    text-align: $textAlign;
+}
+
+.button {
+    @include margins(2px 2px);
```
Comment (kev-chan) [inline src/scss/styles.scss]: consolidate this one

File: src/scss/styles.scss [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            vertical-align: middle;
+            width: 85%;
+
+            .date {
+                @include margins(0px 0px 5px);
+                @include font(false, 12px, false, italic, false, false);
+            }
+
+            .message {
+                @include margins(4px 0px 0px);
```
Comment (kev-chan) [inline src/scss/styles.scss]: why are you specifying margins when all you are using is margin-top?

File: src/scss/styles.scss [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            }
+        }
+
+        .tweetInfo {
+            display: inline-block;
+            vertical-align: middle;
+            width: 85%;
+
+            .date {
+                @include margins(0px 0px 5px);
```
Comment (kev-chan) [inline src/scss/styles.scss]: why are you specifying margins when all you are using is margin-bottom?

File: src/scss/styles.scss [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        border-bottom: 1px solid #d1d1d1;
+
+        .userInfo {
+            text-align: center;
+            display: inline-block;
+            width: 15%;
+            vertical-align: middle;
+
+            .userName {
+                @include margins(0px 0px 5px);
```
Comment (kev-chan) [inline src/scss/styles.scss]: why are you specifying margins when all you are using is margin-top?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/scss/styles.scss (hunk 4 above)
Comment (dhagandev) [reply src/scss/styles.scss]: By that logic maybe I shouldnt be including margins for line 17?

**Comment by kev-chan** [comment]
your scss is generating bad css, please take a look

File: src/scss/styles.scss [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    @include margins(8px auto);
+    border: solid 1px;
+    width: 90%;
+
+    .wait {
+        @include font($defaultFontFamily, 24px, $defaultWeight, italic, $defaultColor, center);
+    }
+
+    .error {
+        @include font($defaultFontFamily, 24px, $defaultWeight, $defaultStyle, #ff0000, center);
```
Comment (kev-chan) [inline src/scss/styles.scss]: why the choice of explicitly stating the default style vs just allowing the css to use the default naturally?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/scss/styles.scss (hunk 7 above)
Comment (dhagandev) [reply src/scss/styles.scss]: I tried to see how to have those as optional, the closest I saw were if $variable { set the field } and when I tried that, it was not setting the field when I did have values.

File: src/scss/styles.scss [hunk 8]
```diff
@@ -0,0 +1,119 @@
+@mixin margins($values...) {    
+    @each $var in $values {
+        margin: #{$var};
+    }
+}
+
+@mixin font($family, $size, $weight, $style, $color, $textAlign) {
```
Comment (kev-chan) [inline src/scss/styles.scss]: this is confusing
better to have smaller multiple mixins or make a smaller mixin and keep the other styles specific to the areas they are needed

**Review by kev-chan — CHANGES_REQUESTED** [review]

**Pull request merged**